packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -58,14 +72,44 @@ export class EditorSettingsSidebarComponent {
 	 * @param {EditorSidebarSection} sectionName Name of section.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async expandSectionIfCollapsed( sectionName: EditorSidebarSection ): Promise< void > {
+	async expandSection( sectionName: EditorSidebarSection ): Promise< void > {
 		// Avoid the wpcalypso/staging banner
 		await this.scrollToBottomOfSidebar();
 		if ( ! ( await this.frame.isVisible( selectors.expandedSection( sectionName ) ) ) ) {
 			await this.frame.click( selectors.sectionToggle( sectionName ) );
 		}
 	}
 
+	/* Status & Visibility */
+
+	/**
+	 * Sets the post visibility to the provided visibility setting.
+	 *
+	 * @param {PostVisibilityOptions} visibility Desired post visibility setting.
+	 */
+	async setVisibility( visibility: PostVisibilityOptions ): Promise< void > {
+		await this.expandSection( 'Status & Visibility' );
+		await this.frame.click( selectors.visibilityToggle );
+		const popoverHandle = await this.frame.waitForSelector( selectors.visibilityPopover );
+		await popoverHandle.waitForElementState( 'stable' );
```

Comment:
Clicking on the visibility settings bring up a popover within the iframed editor.